### PR TITLE
Move simple t.Fatalf test assertions to require.NoError

### PR DESCRIPTION
**Pull request by tvolk131**
## Change Description
Many tests use the following assertion template:
```
	if err != nil {
		t.Fatalf("some description: %v", err)
	}
```

which we can replace with:
```
	require.NoError(t, err, "some description")
```

This is functionally identical but more concise and, I would argue, easier to read. Performing this change reduces the size of affected test files by up to 6.5%. Not bad for something so simple!

It's also worth noting that, in order to prevent 80+ character lines, we can choose to only perform this conversion on lines with a certain description length. Looking at the replacement assertion above, and counting the tab as 8 characters as specified in the contribution guide, an assertion with an empty description is 35 characters, allowing for up to 45-character descriptions. This is why the find-and-replace regex (see below) has `[^"%]{0,45}` which allows for a description of between 1 and 45 characters.

## Don't Trust - Verify!
Sifting through all of the changes here would take a long time and be quite tedious and error-prone. It makes more sense to review _how_ I made the changes, reproduce the steps, and compare them to this PR. To do this, perform the following steps:
1. Start a new local branch in the lnd repo, branching from the same commit as this PR
2. Use VS Code's find-and-replace feature *with regex mode enabled* using the following regex:
`\n\tif err != nil \{\n\t\tt.Fatalf\("([^"%]{0,45}): %[a-z]", err\)\n\t\}\n`
and the following replacement:
`\n\trequire.NoError(t, err, "$1")\n`
and the following "files to include":
`**/**test**` (so we only modify test files)
and the following "files to exclude":
`itest,channeldb/migration*` (so we don't modify integration tests or migration code)

For me, this changed 1855 instances across 92 files. There are currently a total of 6090 calls to `t.Fatalf()`, so this accounts for 30.5% of them.
3. Add the `require` dependency (`github.com/stretchr/testify/require`) to files where it's missing - VS Code should automatically add it to most files, but for me it missed `chainntnfs/test_utils.go`, `chainntnfs/bitcoindnotify/bitcoind_test.go`, and `chainntnfs/btcdnotify/btcd_test.go`
4. Commit your changes in the local branch and compare to the commit in this PR using `git diff <commit-id> <commit-id>` - there should be no diffs

### CODE REVIEW

**Review by carlaKC — APPROVED** [review]
Very nice to standardize this across the codebase! 

Needs rebase (which is the problem with PRs of this magnitude), but looks good otherwise.

**Review by bhandras — COMMENTED** [review]
Cool PR, verified using the provided instructions. Please rebase (or redo on a fresh HEAD).

**Comment by tvolk131** [comment]
@bhandras I just pushed a fresh redo!

**Review by bhandras — APPROVED** [review]
Verified the new diff. Interestingly on my end vscode does a few weird additional replaces but it could be because I have an outdated version (using vim normally).

LGTM, thanks @tvolk131 :tada:

**Comment by tvolk131** [comment]
> Interestingly on my end vscode does a few weird additional replaces

That's strange. I just double checked on my machine and I'm not seeing any diffs. I guess if the diffs on your end are small, it's probably not worth worrying about. ¯\\_(ツ)_/¯

File: chainntnfs/btcdnotify/btcd_test.go [hunk 1]
```diff
@@ -109,22 +98,16 @@ func TestHistoricalConfDetailsTxIndex(t *testing.T) {
 	// Now, we'll create a test transaction and attempt to retrieve its
 	// confirmation details.
 	txid, pkScript, err := chainntnfs.GetTestTxidAndScript(harness)
-	if err != nil {
-		t.Fatalf("unable to create tx: %v", err)
-	}
+	require.NoError(t, err, "unable to create tx")
 	if err := chainntnfs.WaitForMempoolTx(harness, txid); err != nil {
```
Comment (yyforyongyu) [inline chainntnfs/btcdnotify/btcd_test.go:102]: nit: looks like this line is not replaced.

File: chainntnfs/test/test_interface.go [hunk 2]
```diff
@@ -39,17 +39,13 @@ func testSingleConfirmationNotification(miner *rpctest.Harness,
 	// We're spending from a coinbase output here, so we use the dedicated
 	// function.
 	txid, pkScript, err := chainntnfs.GetTestTxidAndScript(miner)
-	if err != nil {
-		t.Fatalf("unable to create test tx: %v", err)
-	}
+	require.NoError(t, err, "unable to create test tx")
 	if err := chainntnfs.WaitForMempoolTx(miner, txid); err != nil {
 		t.Fatalf("tx not relayed to miner: %v", err)
```
Comment (yyforyongyu) [inline chainntnfs/test/test_interface.go:44]: here too.

File: chainntnfs/test/test_interface.go [hunk 3]
```diff
@@ -113,17 +105,13 @@ func testMultiConfirmationNotification(miner *rpctest.Harness,
 	// Again, we'll begin by creating a fresh transaction, so we can obtain
 	// a fresh txid.
 	txid, pkScript, err := chainntnfs.GetTestTxidAndScript(miner)
-	if err != nil {
-		t.Fatalf("unable to create test addr: %v", err)
-	}
+	require.NoError(t, err, "unable to create test addr")
 	if err := chainntnfs.WaitForMempoolTx(miner, txid); err != nil {
 		t.Fatalf("tx not relayed to miner: %v", err)
```
Comment (yyforyongyu) [inline chainntnfs/test/test_interface.go:110]: and here.

File: chainntnfs/test/test_interface.go [hunk 4]
```diff
@@ -312,9 +294,7 @@ func testSpendNotification(miner *rpctest.Harness,
 
 	// Broadcast our spending transaction.
 	spenderSha, err := miner.Client.SendRawTransaction(spendingTx, true)
-	if err != nil {
-		t.Fatalf("unable to broadcast tx: %v", err)
-	}
+	require.NoError(t, err, "unable to broadcast tx")
 
 	if err := chainntnfs.WaitForMempoolTx(miner, spenderSha); err != nil {
 		t.Fatalf("tx not relayed to miner: %v", err)
```
Comment (yyforyongyu) [inline chainntnfs/test/test_interface.go:300]: same here.

File: chainntnfs/test/test_interface.go [hunk 5]
```diff
@@ -464,9 +440,7 @@ func testMultiClientConfirmationNotification(miner *rpctest.Harness,
 	// We'd like to test the case of a multiple clients registered to
 	// receive a confirmation notification for the same transaction.
 	txid, pkScript, err := chainntnfs.GetTestTxidAndScript(miner)
-	if err != nil {
-		t.Fatalf("unable to create test tx: %v", err)
-	}
+	require.NoError(t, err, "unable to create test tx")
 	if err := chainntnfs.WaitForMempoolTx(miner, txid); err != nil {
 		t.Fatalf("tx not relayed to miner: %v", err)
```
Comment (yyforyongyu) [inline chainntnfs/test/test_interface.go:445]: same here.

File: chainntnfs/test/test_interface.go [hunk 6]
```diff
@@ -535,9 +507,7 @@ func testTxConfirmedBeforeNtfnRegistration(miner *rpctest.Harness,
 	// spending from a coinbase output here, so we use the dedicated
 	// function.
 	txid3, pkScript3, err := chainntnfs.GetTestTxidAndScript(miner)
-	if err != nil {
-		t.Fatalf("unable to create test tx: %v", err)
-	}
+	require.NoError(t, err, "unable to create test tx")
 	if err := chainntnfs.WaitForMempoolTx(miner, txid3); err != nil {
 		t.Fatalf("tx not relayed to miner: %v", err)
```
Comment (yyforyongyu) [inline chainntnfs/test/test_interface.go:512]: same here.

File: chainntnfs/test/test_interface.go [hunk 7]
```diff
@@ -548,36 +518,26 @@ func testTxConfirmedBeforeNtfnRegistration(miner *rpctest.Harness,
...
-	}
+	require.NoError(t, err, "unable to generate block")
 
 	txid1, pkScript1, err := chainntnfs.GetTestTxidAndScript(miner)
-	if err != nil {
-		t.Fatalf("unable to create test tx: %v", err)
-	}
+	require.NoError(t, err, "unable to create test tx")
 	if err := chainntnfs.WaitForMempoolTx(miner, txid1); err != nil {
 		t.Fatalf("tx not relayed to miner: %v", err)
```
Comment (yyforyongyu) [inline chainntnfs/test/test_interface.go:526]: +1

File: chainntnfs/test/test_interface.go [hunk 8]
```diff
@@ -548,36 +518,26 @@ func testTxConfirmedBeforeNtfnRegistration(miner *rpctest.Harness,
...
 	}
 
 	txid2, pkScript2, err := chainntnfs.GetTestTxidAndScript(miner)
-	if err != nil {
-		t.Fatalf("unable to create test tx: %v", err)
-	}
+	require.NoError(t, err, "unable to create test tx")
 	if err := chainntnfs.WaitForMempoolTx(miner, txid2); err != nil {
 		t.Fatalf("tx not relayed to miner: %v", err)
```
Comment (yyforyongyu) [inline chainntnfs/test/test_interface.go:532]: +1

File: chainntnfs/test/test_interface.go [hunk 9]
```diff
@@ -725,17 +675,13 @@ func testLazyNtfnConsumer(miner *rpctest.Harness,
 	// Create a transaction to be notified about. We'll register for
 	// notifications on this transaction but won't be prompt in checking them
 	txid, pkScript, err := chainntnfs.GetTestTxidAndScript(miner)
-	if err != nil {
-		t.Fatalf("unable to create test tx: %v", err)
-	}
+	require.NoError(t, err, "unable to create test tx")
 	if err := chainntnfs.WaitForMempoolTx(miner, txid); err != nil {
 		t.Fatalf("tx not relayed to miner: %v", err)
```
Comment (yyforyongyu) [inline chainntnfs/test/test_interface.go:680]: +1

File: chainntnfs/test/test_interface.go [hunk 10]
```diff
@@ -768,17 +712,13 @@ func testLazyNtfnConsumer(miner *rpctest.Harness,
 	// if the first transaction has confirmed doesn't mean that we shouldn't
 	// be able to see if this transaction confirms first
 	txid, pkScript, err = chainntnfs.GetTestTxidAndScript(miner)
-	if err != nil {
-		t.Fatalf("unable to create test tx: %v", err)
-	}
+	require.NoError(t, err, "unable to create test tx")
 	if err := chainntnfs.WaitForMempoolTx(miner, txid); err != nil {
 		t.Fatalf("tx not relayed to miner: %v", err)
```
Comment (yyforyongyu) [inline chainntnfs/test/test_interface.go:717]: +1

File: chainntnfs/test/test_interface.go [hunk 11]
```diff
@@ -829,35 +767,27 @@ func testSpendBeforeNtfnRegistration(miner *rpctest.Harness,
...
 
 	// We'll then spend this output and broadcast the spend transaction.
 	spendingTx := chainntnfs.CreateSpendTx(t, outpoint, output, privKey)
 	spenderSha, err := miner.Client.SendRawTransaction(spendingTx, true)
-	if err != nil {
-		t.Fatalf("unable to broadcast tx: %v", err)
-	}
+	require.NoError(t, err, "unable to broadcast tx")
 	if err := chainntnfs.WaitForMempoolTx(miner, spenderSha); err != nil {
 		t.Fatalf("tx not relayed to miner: %v", err)
```
Comment (yyforyongyu) [inline chainntnfs/test/test_interface.go:777]: +1

File: chainntnfs/test_utils.go [hunk 12]
```diff
@@ -181,9 +176,7 @@ func NewMiner(t *testing.T, extraArgs []string, createChain bool,
 	extraArgs = append(extraArgs, trickle)
 
 	node, err := rpctest.New(NetParams, nil, extraArgs, "")
-	if err != nil {
-		t.Fatalf("unable to create backend node: %v", err)
-	}
+	require.NoError(t, err, "unable to create backend node")
 	if err := node.SetUp(createChain, spendableOutputs); err != nil {
 		node.TearDown()
 		t.Fatalf("unable to set up backend node: %v", err)
```
Comment (yyforyongyu) [inline chainntnfs/test_utils.go:182]: +1

**Comment by yyforyongyu** [comment]
> I'll follow up with another PR that's pretty similar to this one.

That'll be dope!

> Do you think it's worth making a tracking bug for t.Fatalf migration?

Not sure what you meant, like open a new issue to track everything? Don't think it's a bug tho, but rather a nice cleanup.

**Comment by tvolk131** [comment]
> Not sure what you meant, like open a new issue to track everything?

Sorry! I used Google lingo by mistake - yes, I meant a new issue.

**Comment by tvolk131** [comment]
@bhandras any chance we can get this merged before there are more conflicts?

**Comment by guggero** [comment]
We'll merge this first thing once 0.15 is out.

I think your regular expression can be simplified to `\n\tif err != nil \{\n\t\tt.Fatalf\("([^"%]+): %v", err\)\n\t\}\n` which also matches a few more occurrences.

**Comment by tvolk131** [comment]
> We'll merge this first thing once 0.15 is out.

Gotcha! My apologies, I didn't realize there was a blocker for merging this.

> I think your regular expression can be simplified to `\n\tif err != nil \{\n\t\tt.Fatalf\("([^"%]+): %v", err\)\n\t\}\n` which also matches a few more occurrences.

Ahh good call! I tweaked the regex slightly from what you have there, rebased with that new regex, and updated the PR description with that regex as well as up-to-date numbers for instance counts, percentages, etc...

**Review by guggero — APPROVED** [review]
Ran the regex myself, LGTM :tada: 

@tvolk131 since this is test code only, we could get it in for 0.15 during the RC phase. But then you'd need to update the release notes (move from 0.16 to 0.15).

**Review by yyforyongyu — APPROVED** [review]

**Comment by tvolk131** [comment]
> we could get it in for 0.15 during the RC phase. But then you'd need to update the release notes (move from 0.16 to 0.15

Done!

**Comment by guggero** [comment]
Affects unit tests only, merging for 0.15.0.

**Pull request merged**